### PR TITLE
feat: Add event listener trackers to Map and Label

### PR DESCRIPTION
**Pull request by RODO94**
## What does this PR do?

This PR begins to add event listeners and map dispatches to the public component of ``MapAndLabel`` which can then lay the foundations for Work to Trees and drawing shapes on a map and tracking them with List components. 

- [x] Track Geometry and Area
- [x] Add light validation 
- [x] Generate a dummy card with Geo data

### CODE REVIEW

**Comment by RODO94** [comment]
[Allow applicants to 'work to trees' to see and add more information about protected trees on a map (separate component)](https://trello.com/c/JafdzU94/2855-allow-applicants-to-work-to-trees-to-see-and-add-more-information-about-protected-trees-on-a-map-separate-component)

File: editor.planx.uk/src/@planx/components/MapAndLabel/Public.tsx [hunk 1]
```diff
@@ -1,17 +1,62 @@
...
 import { PublicProps } from "../ui";
 import { MapAndLabel } from "./model";
 
 type Props = PublicProps<MapAndLabel>;
 
+export type Boundary = Feature | undefined;
+
 function MapAndLabelComponent(props: Props) {
   const teamSettings = useStore.getState().teamSettings;
+  const [boundary, setBoundary] = useState<Boundary>();
```
Comment (jessicamcinchak) [inline editor.planx.uk/src/@planx/components/MapAndLabel/Public.tsx:20]: nit: for simplicity, I wouldn't export a type `Boundary` (unless there's a use case outside of this file??), I'd do this all in the same line like so: 
```
const [boundary, setBoundary] = useState<Feature | undefined>()
```

File: editor.planx.uk/src/@planx/components/MapAndLabel/Public.tsx [hunk 2]
```diff
@@ -1,17 +1,62 @@
...
+  const [area, setArea] = useState<number | undefined>(0);
+  const [objectArray, setObjectArray] = useState<any>([]);
```
Comment (jessicamcinchak) [inline editor.planx.uk/src/@planx/components/MapAndLabel/Public.tsx:22]: I'm not sure I understand the purpose of `objectArray` here (variable name is quite vague & it's never rendered)? Is boundary sufficient alone?

**Review by jessicamcinchak — COMMENTED** [review]
Overall looking very good and working as expected ! 

Your PR description mentions "light validation" which I'd expect to see implemented as an `<ErrorWrapper>` around the `<MapContainer>` with a validation rule (for example, at least one feature must be drawn to "Continue" - see DrawBoundary for existing example of this). Does that sound like the right track or are you thinking about validation differently here?

File: editor.planx.uk/src/@planx/components/MapAndLabel/Public.tsx (hunk 2 above)
Comment (RODO94) [reply editor.planx.uk/src/@planx/components/MapAndLabel/Public.tsx:23]: It's a way for me to track the event listener dispatching events and collecting info into an array. Most likely negated in a PR or two when multiple items get added to the map and we can use the geoJSON features, but helpful for now if I wanted to test adding a list of items to the Public comp... is it too rough to add in the PR? @jessicamcinchak

Comment (jessicamcinchak) [reply editor.planx.uk/src/@planx/components/MapAndLabel/Public.tsx:23]: Thanks for explanation - happy for it to merge here now better understanding it's work-in-progress :slightly_smiling_face:

**Review by jessicamcinchak — APPROVED** [review]

**Pull request merged**